lib/actions/destroy.js

```diff
@@ -13,25 +13,18 @@ const Boom = require('boom');
  */
 module.exports = function destroyOneRecord(route, options) {
 
-    return (request, reply) => {
+    return async (request, h) => {
 
         const actionUtil = require('../actionUtil')(request, options);
 
         const Model = actionUtil.parseModel();
         const keys = actionUtil.getKeys();
 
-        Model.query().deleteById(keys.parent.value)
-        .asCallback((error, rowsDeleted) => {
+        const rowsDeleted = await Model.query().deleteById(keys.parent.value);
 
-            if (error) {
-                return reply(error);
-            }
-
-            if (rowsDeleted === 1) {
-                return reply().code(204);
-            }
-
-            return reply(Boom.notFound('Record not found'));
-        });
+        if (rowsDeleted === 0) {
+            return Boom.notFound('Record not found');
```

Comment:
what are your thoughts on returning a 204 status code here?

asking because I got confused about how to handle this case while writing a customized delete handler today. I mention 204 from the reasoning here: http://eloquentjavascript.net/20_node.html#p_nmATyMSIB9

but a 404 seems to make just as much, if not more, sense here.

anyways, food for thought, the only resolution I reached was getting confused 👍

Reply:
This is a borderline philosophical discussion, but (to me) it _feels_ like 404 is correct in that it provides useful feedback to the caller - e.g. the url requested `DELETE /model/{id}` doesn't exist.  I get the idempotence argument, but (again, to me) that feels like it's sacrificing utility on the altar of a largely arbitrary ideal.

tl;dr: You could convincingly argue in favor of either code, but I'm driving this bus, so it's a 404 :trollface:

Reply:
👏 👏  cool cool, that's all clear. thanks for the thoughts, dude 🙏

Reply:
I believe github's API handles it as a 404, which is what bedwetter originally based its conventions on.  I am a big fan of idempotence– even wrote about it in the #developers channel a month or two ago– but can see it both ways here.  I can really see it both ways!! 
 Given Matt's stance, I think the next thing to do would be to look into how the web clients in our frontend projects can easily ignore 404s if that's desired.  Of course one issue with that is distinguishing between a missing resource versus hitting an endpoint that doesn't exist.  Usually the API will provide different `message`s when the 404 is for a resource versus an endpoint.